### PR TITLE
Switch bottom sheet menu items based on BoxObject type

### PR DESCRIPTION
**Pull request by L-Henke**
Instead of a previously proposed instanceof switch, the interface
BottomSheetMenu is added to BoxObject. With this the subclass can define
which menu should be shown in the bottom sheet.

### CODE REVIEW

File: qabelbox/src/main/res/menu/box_folder_bottom_sheet.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:id="@+id/open" android:title="@string/Open" android:icon="@drawable/ic_share_black"/>
+    <item android:id="@+id/share" android:title="@string/Share" android:icon="@drawable/ic_share_black"/>
```
Comment (dannywerner) [inline qabelbox/src/main/res/menu/box_folder_bottom_sheet.xml]: we don't have a folder share

**Comment by dannywerner** [comment]
- pls add a bottom sheet layout for boxexternal files with handling of shared state (dettach or delete)
- extends BoxFile.getBottomSheetMenuResId to return the correct bottom sheet (unshare if needed, share if possibility (what is with received share files... can this share to another?)...

File: qabelbox/src/main/res/menu/box_folder_bottom_sheet.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:id="@+id/open" android:title="@string/Open" android:icon="@drawable/ic_share_black"/>
```
Comment (dannywerner) [inline qabelbox/src/main/res/menu/box_folder_bottom_sheet.xml]: open is imho incorrect. with the open command, the app launch a external app to show this file, but this is a directory

**Comment by L-Henke** [comment]
I forgot, that the bottom sheets are having some dynamic parts (share/unshare based on share status). I'll think about a better solution.

**Comment by dannywerner** [comment]
thanks, but its a blocker, iam want to finish the share features

**Comment by dannywerner** [comment]
pls REMOVE pr if not support a solution

**Comment by jmt85** [comment]
what is the condition of this pr?

**Comment by jmt85** [comment]
please merge master into the branch

**Pull request closed without merging**